src/histolab/slide.py

```diff
@@ -352,13 +454,28 @@ def _resample(self, scale_factor: int = 32) -> Tuple[PIL.Image.Image, np.array]:
         """
 
         _, _, new_w, new_h = self._resampled_dimensions(scale_factor)
-        level = self._wsi.get_best_level_for_downsample(scale_factor)
-        whole_slide_image = self._wsi.read_region(
-            (0, 0), level, self._wsi.level_dimensions[level]
-        )
+        if self._use_largeimage:
+            mg = "magnification"
+            kws = (
```

Comment:
For clarity and to be explicit I would use meaningful variable name, cause `mg` and `kws` can be misinterpreted in future